src/Services/Git/GitonomyClient.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace fortrabbit\Copy\Services\Git;
+
+use fortrabbit\Copy\Exceptions\GitException;
+use Gitonomy\Git\Admin;
+use Gitonomy\Git\Exception\ProcessException;
+use Gitonomy\Git\Reference\Branch;
+use Gitonomy\Git\Repository;
+use LogicException;
+use RuntimeException;
+
+class GitonomyClient implements Client
+{
+
+	private Repository $repository;
+
+	private string $directory;
+
+	public function clone(
+		string $repository,
+		?string $directory = null,
+		array $options = []
+	) {
+		$this->directory = $directory;
+		$this->repository = Admin::cloneTo($directory, $repository, true, $options);
+	}
+
+	public function push(string $upstream, string $branch = 'master'): string
+	{
+		return $this->run(GitCommand::PUSH, [$upstream, $branch]);
+	}
+
+	public function pull(string $upstream, string $branch = 'master'): string
+	{
+		return $this->run(GitCommand::PULL, [$upstream, $branch]);
+	}
+
+	public function getLocalHead(): ?string
+	{
+		$head = $this->repository->getHead();
+		if ($head instanceof Branch) {
+			return $head->getName();
+		}
+		return $head->getFullName();
+	}
+
+	public function getLocalBranches(): array
+	{
+		$localBranches = array_filter(
+			$this->repository->getReferences()->getBranches(),
+			fn ($branch) => $branch->isLocal()
+		);
+
+		return array_map(function ($branch) {
+			return $branch->getName();
+		}, $localBranches);
+	}
+
+	public function getRemotes(?string $for = 'push'): array
+	{
+		if (!in_array($for, ['push', 'pull'], true)) {
+			throw new LogicException(
+				sprintf(
+					'Argument 1 passed to %s must be "pull" or "push", %s given.',
+					'fortrabbit\Copy\Services\Git\Client::getRemotes()',
```

Comment:
please update

Reply:
I have pushed an update that references the correct class.